### PR TITLE
Add tutorial for adding a map title

### PR DESCRIPTION
**Pull request by willschlitzer**
Adding tutorial for adding a figure title, as requested in  #335.

### CODE REVIEW

File: examples/tutorials/frames.py [hunk 1]
```diff
@@ -36,14 +36,37 @@
...
+fig.show()
+
+########################################################################################
+# To use a title with multiple words, the title must be placed inside another set of
+# quotation marks. To prevent the quotation marks from appearing in the figure title,
+# the ``"\""`` escape character notation can be used to cancel out the quotation marks.
+
+fig = pygmt.Figure()
+fig.coast(shorelines="1/0.5p", region="TT", projection="M25c")
+fig.basemap(frame=["a", "+t\"Trinidad and Tobago\""])
```
Comment (weiji14) [inline examples/tutorials/frames.py:71]: An alternative way (which avoids backslashes) is to use single-quotes `'` outside, with double-quotes `"` inside. See also #247 (which we really need to figure out a proper solution for).

```suggestion
fig.basemap(frame=["a", '+t"Trinidad and Tobago"'])
```

**Review by weiji14 — COMMENTED** [review]
Big thanks for doing this (and all your other work around PyGMT) @willschlitzer! Always nice to have some fresh eyes on our documentation. Just one minor comment on the backslash syntax for now.

**Comment by willschlitzer** [comment]
Could I get one of the maintainers to look at my most recent commit? I can't figure out why it failed so many tests and deployment.

**Comment by seisman** [comment]
I think it's unrelated to your PR. I just re-triggered the CI jobs.

File: examples/tutorials/frames.py [hunk 2]
```diff
@@ -36,14 +36,57 @@
...
+# all primary axes, which the default is all sides of the figure. To designate only
+# some of the axes as primary, an argument that capitlizes only the primary axes can be
+# passed, which is ``"WSne"`` in the example below. The letters correspond with
+# west (left), south (bottom), north (top), and east (right) sides of a figure.
+#
+# The example below used a Cartesian projection, as GMT does not allow axis labels to
+# be set for geographic maps.
+
+fig = pygmt.Figure()
+fig.basemap(region=[0, 10, 0, 20], projection="X4i/3i",
```
Comment (seisman) [inline examples/tutorials/frames.py:90]: Please change them to SI units.

File: examples/tutorials/frames.py [hunk 3]
```diff
@@ -36,14 +36,57 @@
...
+########################################################################################
+# Title
+# -----
+#
+# The figure title can be set by passing ``"+t<title>"`` to the ``frame`` parameter of
+# :meth:`pygmt.Figure.basemap`. Passing multiple arguments to ``frame`` can be done by
+# using a list, as show in the example below.
+
+fig = pygmt.Figure()
+fig.coast(shorelines="1/0.5p", region="IS", projection="M25c")
```
[the remark this answers is not in the document]
Comment (willschlitzer) [reply examples/tutorials/frames.py:61]: I left a comment that fully explained the Iceland code, and a brief explanation for Trinidad and Tobago

File: examples/tutorials/frames.py (hunk 2 above)
Comment (willschlitzer) [reply examples/tutorials/frames.py]: Changed to SI units

File: examples/tutorials/frames.py [hunk 4]
```diff
@@ -36,14 +36,57 @@
...
-fig.coast(shorelines="1/0.5p", region=[-180, 180, -60, 60], projection="M10i")
+fig.coast(shorelines="1/0.5p", region=[-180, 180, -60, 60], projection="M25c")
 fig.basemap(frame="ag")
 fig.show()
+
+########################################################################################
+# Title
+# -----
+#
+# The figure title can be set by passing ``"+t<title>"`` to the ``frame`` parameter of
```
[the remark this answers is not in the document]
Comment (willschlitzer) [reply examples/tutorials/frames.py]: Should I use that syntax for title as you posted it? I put it in my most recent commit but I feel like it doesn't indicate that "+t" is verbatim and "title" is the editable part.

File: examples/tutorials/frames.py [hunk 5]
```diff
@@ -36,14 +36,60 @@
...
-fig.coast(shorelines="1/0.5p", region=[-180, 180, -60, 60], projection="M10i")
+fig.coast(shorelines="1/0.5p", region=[-180, 180, -60, 60], projection="M25c")
 fig.basemap(frame="ag")
 fig.show()
+
+########################################################################################
+# Title
+# -----
+#
+# The figure title can be set by passing ``**+t**\ *title*`` to the ``frame`` parameter of
```
Comment (seisman) [inline examples/tutorials/frames.py:55]: What I meant is:
```suggestion
# The figure title can be set by passing **+t**\ *title* to the ``frame`` parameter of
```

File: examples/tutorials/frames.py [hunk 6]
```diff
@@ -36,14 +36,60 @@
...
+# region="TT" specifies Trinidad and Tobago
+fig.coast(shorelines="1/0.5p", region="TT", projection="M25c")
+fig.basemap(frame=["a", '+t"Trinidad and Tobago"'])
+fig.show()
+
+########################################################################################
+# Axis labels
+# -----------
+#
+# Axis labels can be set by passing ``**x+l**\ *label*`` (or starting with y if
```
Comment (seisman) [inline examples/tutorials/frames.py:81]:
```suggestion
# Axis labels can be set by passing **x+l**\ *label* (or starting with y if
```

**Comment by seisman** [comment]
@willschlitzer 

It looks good now. Please do more things:

1. Merge the master branch, by clicking the "Update branch" button.
2. Fix the styling issue. You can run `make format` locally, or you can just write `/format` in the first line of a comment to automatically format the codes (a feature introduced in #646).

**Review by seisman — APPROVED** [review]

**Pull request merged**